pandas/core/internals/blocks.py

```diff
@@ -1447,8 +1447,18 @@ def quantile(self, qs, interpolation='linear', axis=0):
         -------
         Block
         """
-        values = self.get_values()
-        values, _ = self._try_coerce_args(values, values)
+        if self.is_datetimetz:
```

Comment:
this is getting super messy

Reply:
None of the branches I have in progress would help here.

Allowing for DatetimeArray to be reshaped to (1, nrows) would.

Reply:
add a TODO here